### PR TITLE
Rosidl cli types with `specs_set` fix

### PR DESCRIPTION
**Pull request by InvincibleRMC**
Fixes problem found in #830. Adds a proper default of `None` to `specs_set`. A redo of #826.

### CODE REVIEW

File: rosidl_cli/rosidl_cli/entry_points.py [hunk 1]
```diff
@@ -12,38 +12,41 @@
...
+    :param specs: an optional collection of entry point names to retrieve
+    :param strict: whether to raise or warn on error
     :returns: mapping from entry point names to ``EntryPoint`` instances
-    :rtype: dict
     """
     if specs is not None:
-        specs = set(specs)
+        specs_set = set(specs)
+    else:
+        specs_set = None
```
Comment (fujitatomoya) [inline rosidl_cli/rosidl_cli/entry_points.py:37]: this is the fix for the regression reported with https://github.com/osrf/buildfarm-tools/issues/99, and this is the only difference from https://github.com/ros2/rosidl/pull/826/commits

**Review by fujitatomoya — APPROVED** [review]
lgtm with green CI

**Comment by fujitatomoya** [comment]
Pulls: ros2/rosidl#831
Gist: https://gist.githubusercontent.com/fujitatomoya/ce94ef448780cc49f11d452b9a487972/raw/91a447ce7e711d4075449cc8c6321ad2be8d6132/ros2.repos
BUILD args:  --packages-above-and-dependencies rosidl_cli
TEST args:  --packages-above rosidl_cli
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14672
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21960/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16224/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1474/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22688/)

**Comment by sloretz** [comment]
Pulls: https://github.com/ros2/rosidl/pull/831
Gist: https://gist.githubusercontent.com/sloretz/aff73674edff8eba0caa5b6443bf929b/raw/91a447ce7e711d4075449cc8c6321ad2be8d6132/ros2.repos
BUILD args:  --packages-above-and-dependencies rosidl_cli
TEST args:  --packages-above rosidl_cli
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14751
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22054/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16310/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1572/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22784/)

**Review by sloretz — APPROVED** [review]
Thank you for the PR!

**Pull request merged**